mypy/util.py

```diff
@@ -769,9 +769,13 @@ def format_error(
         return self.style(msg, "red", bold=True)
 
 
-def is_typeshed_file(file: str) -> bool:
+def is_typeshed_file(typeshed_dir: str | None, file: str) -> bool:
+    if typeshed_dir is None:
+        typeshed = TYPESHED_DIR
+    else:
+        typeshed = os.path.abspath(typeshed_dir)
```

Comment:
This might be pretty slow if using a custom typeshed dir, since we are calculating the abspath over and over again. Maybe we should precalculate this in options and pass the absolute path instead?

Reply:
Hm, yeah, good idea. I can do this.